callback-heaven.lisp

```diff
@@ -224,6 +237,20 @@ Note that this memory is not further managed!"
                                           (cffi:translate-name-to-foreign (api-group-name api-group) nil))
            :function-index function-index))))
 
+(defun emit-documentation (documentation stream)
+  (alexandria:when-let ((lines (and documentation
+                                    (uiop:split-string documentation :separator '(#\Newline)))))
+    ;; No need for an (ENDP LINES) clause here. LINES must be non-NULL due to the WHEN-LET above.
+    (cond ((endp (rest lines))
+           (format stream "/** ~A */~%" (first lines)))
+          (t
+           (format stream "/**~%")
+           (dolist (line lines)
+             (if (string= "" line)
+                 (format stream " *~%")
+                 (format stream " * ~A~%" line)))
+           (format stream " */~%")))))
```

Comment:
I must be getting old because the temptation to write this as

```lisp
(format stream "/**~%~{ *~@[ ~A~]~%~} */~%" (substitute nil "" lines :test #'string=))
```

instead was barely noticeable